templates/common/.devcontainer/devcontainer.json/python/devcontainer.json

```diff
@@ -18,7 +18,7 @@
             "nodeGypDependencies": false
         },
         "ghcr.io/devcontainers/features/python:1": {
-            "version": "os-provided"
+            "version": "3.9"
```

Comment:
Can you change the container to base on:

mcr.microsoft.com/devcontainers/python:3.9

Instead of base, to see if it is smaller/faster?

Reply:
Just want to clarify the idea:

If we change the container to "mcr.microsoft.com/devcontainers/python:3.9", then do we actually still want the Python feature? It doesn't seem like that's the intention of the feature - see my discussion with their team at https://github.com/devcontainers/features/issues/430 where I was trying to understand "os-provided".

Also, if we change the container to "mcr.microsoft.com/devcontainers/python:3.9", I presume we'd make the "python:3.9" an ARG to get overriden in devcontainer.json, for easier customization of each template?

I'm experimenting with that base of 3.9 now.

Reply:
Yeah, if python base then you wouldn't need the feature.  Features are always slower, so might make sense, but wanted to see if faster, smaller if we do so.

Reply:
Okay, here are the stats I found on my local M1.

I compared 
`FROM mcr.microsoft.com/devcontainers/python:3.9`
to
`FROM mcr.microsoft.com/vscode/devcontainers/base:0-bullseye`

|                    | python:3.9| base:0-bullseye + python os-provided | + python 3.9 |
|--------------------|-----------|-------------|------------|
| Base image size    | 1.29 GB    | 790.8 MB    | (same) |
| Dev Container size | 1.29 GB    | 1.12 GB     | 1.62 GB | 
| Time to build      | 12282 ms   | 59914 ms    | 94643 ms |

Observations:

* The Python 3.9 image size starts off larger than the bullseye image. The bulls eye + os-provided image is still smaller, but takes much more time to build. It doesn't seem safe to assume the OS has python, however, so I compared to python 3.9 feature, and then the image *and* build time are larger.
* The python image is really fast to build, comparatively!
* I tried adding on extensions to python 3.9 base image, and it didn't seem to change that size of the dev container image, so I think it installs them outside the image? Or maybe it wasn't affected because I already had those extensions locally. So I'm not 100% sure on effect of extensions, there are too many variables there.

So if we're going for faster loads, using a base python 3.9 image seems the way to go. That's a larger change though, since that means changing the other languages too, and setting it up so that an ARG changes the whole base image, not just the tag.

Reply:
I've changed all the devcontainers to use the language-specific base image and not the feature. I've tested that they each successfully load (i.e. no issues creating the dev container) by trying a sample in each language with these changes. I haven't successfully gotten VS Code "Debug" working in every language, but I also don't know if "Debug" worked before my changes.

Reply:
Actually, it seems like I have gotten "Debug" working in them, at least to the point of it prompting to create a new azd environment. So it looks like the necessary debuggers are installed correctly.